src/os/bluestore/BlueStore.h

```diff
@@ -569,10 +573,13 @@ class BlueStore : public ObjectStore,
 
   /// a cache (shard) of onodes and buffers
   struct Cache {
+    BlueStore *store;
     std::mutex lock;                ///< protect lru and other structures
 
     static Cache *create(string type);
 
+    void set_store(BlueStore *bs) { store = bs; }
```

Comment:
@xiexingguo Probably I missed something but my suggestion was just to modify create in the following manner:

static Cache *create(string type, PerfCounters* log)
cache = new ...
cache->logger = log
return cache;
}
I don't expect any issues with such code...